### PR TITLE
Navbar

### PR DESCRIPTION
**Pull request by maxmorton42**
contains navbar router controller and db

### CODE REVIEW

File: app/controllers/api/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @categories = Category.all
+  end
+
+  def show
+    @category = Category.find(params[:id])
+  end
+
+  def new
+    @category = Category.new
+  end
```
Comment (spenhappenin) [inline app/controllers/api/categories_controller.rb]: dont need this as we arent using views

File: app/controllers/api/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    redirect_to category_path
+  end
+
+  private
+   def set_category
+    @category = Category.find(params[:id])
+   end
+
+   def category_params
+    params.require(:user).permit(:name)
```
Comment (spenhappenin) [inline app/controllers/api/categories_controller.rb]: I don't think user is going to work here

File: app/controllers/api/items_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  def destroy
+  end
+
+  private
+   def set_category
+   end
+
+   def category_params
+   end
```
Comment (spenhappenin) [inline app/controllers/api/items_controller.rb]: category?

File: app/controllers/api/item_variants_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  def destroy
+  end
+
+  private
+   def set_category
+   end
+
+   def category_params
+   end
```
Comment (spenhappenin) [inline app/controllers/api/item_variants_controller.rb]: category?

File: app/models/item.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class Item < ApplicationRecord
+  belongs_to :category
+  has_many :itemvariants
```
Comment (spenhappenin) [inline app/models/item.rb]: forgot the underscore

File: app/controllers/api/categories_controller.rb (hunk 1 above)
Comment (DMendez49) [reply app/controllers/api/categories_controller.rb]: this should be fixed

File: app/controllers/api/categories_controller.rb (hunk 2 above)
Comment (DMendez49) [reply app/controllers/api/categories_controller.rb]: this should be fixed

File: app/controllers/api/items_controller.rb (hunk 3 above)
Comment (DMendez49) [reply app/controllers/api/items_controller.rb]: this should be fixed

File: app/controllers/api/item_variants_controller.rb (hunk 4 above)
Comment (DMendez49) [reply app/controllers/api/item_variants_controller.rb]: I meant to put item_variants

File: app/models/item.rb (hunk 5 above)
Comment (DMendez49) [reply app/models/item.rb]: fixed it

**Pull request merged**